rllib/algorithms/ppo/torch/ppo_torch_rl_module.py

```diff
@@ -298,17 +310,11 @@ def input_specs_train(self) -> ModelSpec:
             action_dim = self.config.action_space.shape[0]
             action_spec = TorchTensorSpec("b, h", h=action_dim)
 
-        obs_specs = (
-            self.encoder.input_specs() if self.encoder else self.pi.input_specs()
-        )
-        spec = ModelSpec(
-            {
-                SampleBatch.OBS: obs_specs,
-                SampleBatch.NEXT_OBS: obs_specs,
-                SampleBatch.ACTIONS: action_spec,
-            }
-        )
-
+        spec_dict = self.shared_encoder.input_spec()
```

Comment:
in case shared_encoder is IdentityEncoder(), does it actually understand the real specs?

Reply:
it will become a non-restrictive spec (None), the next block will yell at you then.